notes.org

```diff
@@ -37,24 +37,29 @@
 		 let posts_in_blog = query!(Post, &conn, blog = {blog}).load(&conn)
 	 }
 	 #+END_SRC
+
 	 struct Posts
+
+	 #+BEGIN_SRC sql
 	 CREATE TABLE posts (
-  id SERIAL PRIMARY KEY,
-  title VARCHAR NOT NULL,
-  body TEXT NOT NULL,
-  published BOOLEAN NOT NULL DEFAULT 'f'
-)
+		id SERIAL PRIMARY KEY,
+		title VARCHAR NOT NULL,
+		body TEXT NOT NULL,
+		published BOOLEAN NOT NULL DEFAULT 'f'
+	 )
+	 #+END_SRC
+
 * Roadmap
 ** vNext
-	 + foreign key constraints in db, cascade setting
+	 + foreign key constraints in db, cascade setting [[https://github.com/Electron100/butane/issues/21]]
 	 + incremental save
```

Comment:
This one isnt clear to me.
does this mean doing writes immediately when changing a struct instance?  sounds crazy.
or `Many.add` writing immediately?

Reply:
The original intention of this line was `save` being able to do an update with only the fields that have actually changed, as a performance enhancement. However that really doesn't fit well with the current architecture of assigning directly to struct fields and I'm not sure I think it's a good idea any more. I think at this point I'd do something like `DataResult` (or like what Diesel does) to allow updates with a struct containing only a subset of the model fields.